src/python/pants/build_graph/build_file_aliases.py

```diff
@@ -40,6 +41,13 @@ def wrap(cls, context_aware_object_factory, *target_types):
       if not target_types:
         raise ValueError('The given `context_aware_object_factory` {} must expand at least 1 '
                          'produced type; none were registered'.format(context_aware_object_factory))
+      deprecated_conditional(
+        lambda: len(target_types) > 1,
+        '1.7.0.dev0',
+        'TargetMacro.Factory instances that construct more than one type are no longer supported. '
+        'Consider using a `context_aware_object_factory, which can construct any number of '
+        'different objects.'
+      )
```

Comment:
Should this be a separate change?

Reply:
Maybe, but I no longer remember which test I was fixing when I changed it, so I'd rather land it here.